### PR TITLE
Fix integration order for Lobatto basis with square element

### PR DESCRIPTION
**Pull request by dreamer2368**
So far, integration order has been erroneously adjusted when the Gauss-Lobatto basis is used with square element, as below:
```
if (intRuleType == 1 && vfes->GetFE(i)->GetGeomType() == Geometry::SQUARE)
      integrationOrder--;  // when Gauss-Lobatto
```

This somehow passed the consistency test with the time integration. However, it does not pass the convergence test with `utils/compute_rhs`. After this fix, the errors of manufactured solutions significantly decreased (about a factor of 4).

Beside this, another manufactured solution test `mms_euler_2d` is added.

### CODE REVIEW

File: test/inputs/input.reaction.ini [hunk 1]
```diff
@@ -16,10 +16,10 @@ equation_system = 'navier-stokes'
 outdirBase = output
 
 [time]
-cfl = 0.5
+cfl = 0.2
 integrator = rk4
 enableConstantTimestep = True
-dt_fixed = 4.6e-4
+dt_fixed = 2.0e-4
```
Comment (trevilo) [inline test/inputs/input.reaction.ini:22]: Did changing the quadrature lead to having to reduce `dt`?  Just wondering what happened here.

File: test/test_gradient.cpp [hunk 2]
```diff
@@ -0,0 +1,269 @@
+/* A utility to transfer tps solutions from one mesh to another */
```
Comment (trevilo) [inline test/test_gradient.cpp:1]: This looks like the commented out test.  Is it ready to be included?  If so, we should do that.  If not, maybe this should part of a separate PR.

**Review by trevilo — COMMENTED** [review]
Hey @dreamer2368, looks pretty good to me.  Once you eliminate the dead code (rather than comment it out) and decide what you want to do with the gradient test, I'll get it merged.

File: test/inputs/input.reaction.ini (hunk 1 above)
Comment (dreamer2368) [reply test/inputs/input.reaction.ini:22]: It needed a smaller timestep, but did not need to change the reference solution. Didn't look into it much, as the rate is fictitious anyway.

File: test/Makefile.am [hunk 3]
```diff
@@ -176,4 +176,15 @@ if MASA_ENABLED
 test_interp_em_LDFLAGS 	 += $(MASA_LIBS)
 endif
 
+#check_PROGRAMS += test_grad
```
[the remark this answers is not in the document]
Comment (dreamer2368) [reply test/Makefile.am]: Gradient test is now added in the ci test.

File: test/test_gradient.cpp (hunk 2 above)
Comment (dreamer2368) [reply test/test_gradient.cpp:1]: Gradient test is now added in the ci test.

**Pull request merged**